File: app/src/main/java/name/seguri/android/lock/LockAccessibilityTileService.java
package name.seguri.android.lock;

import android.annotation.TargetApi;
import android.content.Intent;
import android.os.Build;
import android.service.quicksettings.TileService;

@TargetApi(Build.VERSION_CODES.P)
public class LockAccessibilityTileService extends TileService {

    @Override
    public void onClick() {
        Intent intent = new Intent(getApplicationContext(), MainActivity.class);
        intent.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP | Intent.FLAG_ACTIVITY_NEW_TASK);
        startActivityAndCollapse(intent);
    }
}

File: app/src/main/java/name/seguri/android/lock/LockAccessibilityTileService.kt
package name.seguri.android.lock

import android.annotation.TargetApi
import android.content.Intent
import android.os.Build
import android.service.quicksettings.TileService
import name.seguri.android.lock.MainActivity

@TargetApi(Build.VERSION_CODES.P)
class LockAccessibilityTileService : TileService() {

    override fun onClick() {
        val intent = Intent(applicationContext, MainActivity::class.java)
        intent.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP or Intent.FLAG_ACTIVITY_NEW_TASK)
        startActivityAndCollapse(intent)
    }
}
